website/pages/docs/connect/ingress_gateway.mdx

```diff
@@ -0,0 +1,66 @@
+---
+layout: docs
+page_title: Connect - Ingress Gateways
+sidebar_title: Ingress Gateways
+description: >-
+  An Ingress Gateway enables ingress traffic from services outside the Consul
+  service mesh to services inside the Consul service mesh. This section details
+  how to use Envoy and describes how you can plug in a gateway of your choice.
+---
+
+# Ingress Gateways
+
+-> **1.8.0+:**  This feature is available in Consul versions 1.8.0 and newer.
+
+Ingress gateways enable ingress traffic from services outside the Consul
+service mesh to services inside the Consul service mesh. These gateways act as
+an entrypoint for outside traffic and allow users to define what services
+should be exposed and on what port. They allow for defining a set of
+[listeners](/docs/agent/config-entries/ingress-gateway#listeners) that each map
+to a set of backing
+[services](/docs/agent/config-entries/ingress-gateway#services).
+
+Depending on the
+[protocol](/docs/agent/config-entries/ingress-gateway#protocol) defined for a
+listener, multiple services can be specified for a single listener. In this
+case, the ingress gateway relies on host/authority headers to decide the
+service that should receive the traffic.
+
+To enable easier service discovery, a new Consul [DNS
+subdomain](/docs/agent/dns#ingress-service-lookups) is provided, on
+`<service>.ingress.<domain>`.
+
+## Prerequisites
+
+Ingress gateways also require that your Consul datacenters are configured correctly:
+
+- You'll need to use Consul version 1.8.0.
+- Consul [Connect](/docs/agent/options#connect) must be enabled in the datacenter.
+- [gRPC](/docs/agent/options#grpc_port) must be enabled.
+
+Currently, Envoy is the only proxy with ingress gateway capabilities in Consul.
+
+## Running and Using an Ingress Gateway
+
+The steps that need to occur for an Ingress Gateway to proxy traffic to services in the Consul mesh are as follows:
+
+1. Run an envoy proxy using the [envoy subcommand](/docs/commands/connect/envoy), specifying the `ingress` gateway type.
+2. Create an `ingress-gateway` [configuration
+entry](/docs/agent/config-entries/ingress-gateway) that defines a set of listeners that expose the desired backing services.
+The config entry can be applied via the [CLI](/docs/commands/config/write) or [API](/api/config#apply-configuration).
+3. Ensure that [intentions](/docs/connect/intentions) are setup between the ingress gateway service and the backing services.
```

Comment:
```suggestion
1. Ensure that [intentions](/docs/connect/intentions) are setup between the ingress gateway service and the backing services.
```
Recently, I've started specifying Consul intention for clarity. Also it seems like we're using three terms for ingress gateways, "ingress gateways", "ingress gateways service", and "ingress gateway proxy". Maybe at the beginning of this doc we can explain that an ingress gateway is a type of proxy and must be registered as a service in Consul? Then moving forward just call it an ingress gateway.

Reply:
The detailed config section below is great for helping users understand ingress gateways as a proxy and the service registration requirments. I definitely think a more concise version at the beginning would be helpful.